mono/mini/mini-runtime.c

```diff
@@ -4564,6 +4564,12 @@ mini_init (const char *filename, const char *runtime_version)
 	mono_profiler_state.context_get_result = mini_profiler_context_get_result;
 	mono_profiler_state.context_free_buffer = mini_profiler_context_free_buffer;
 
+	if (g_hasenv ("MONO_PROFILE")) {
+		gchar *profile_env = g_getenv ("MONO_PROFILE");
```

Comment:
Can't this be set using MONO_ENV_OPTIONS ?

Reply:
That is also handled in driver.c and not in use for embedders, having an env option for embedding use cases (having a different driver than driver.c) is beneficial, similar to what we have for several other areas using explicit MONO_* env variables.

Reply:
Can't an embedder call mono_jit_parse_options () with "--profile=" ?

Reply:
That option is added by this PR. There are some scenarios where the env still make sense since it simplify a couple of launch scenarios. If we don't add the support in this PR, then I will need to implement things downstream to do same thing, totally doable, but thought it made sense to add it into this PR and handle it similar to other MONO_* env variables.